### PR TITLE
faet: leaderboard filter by char class

### PR DESCRIPTION
**Pull request by or2e**
image

### CODE REVIEW

File: src/WebApi/Controllers/LeaderboardController.cs [hunk 1]
```diff
@@ -1,9 +1,8 @@
 using Crpg.Application.Characters.Models;
 using Crpg.Application.Characters.Queries;
 using Crpg.Application.Common.Results;
-using Crpg.Application.Games.Commands;
```
Comment (or2e) [inline src/WebApi/Controllers/LeaderboardController.cs:4]: unused

File: src/WebApi/Controllers/LeaderboardController.cs [hunk 2]
```diff
@@ -19,11 +18,15 @@ public class LeaderboardController : BaseController
     /// <response code="200">Ok.</response>
     [HttpGet("leaderboard")]
     [ResponseCache(Duration = 1 * 60 * 1)] // 1 minutes
-    public Task<ActionResult<Result<IList<CharacterPublicViewModel>>>> GetLeaderboard([FromQuery] Region? region)
+    public Task<ActionResult<Result<IList<CharacterPublicViewModel>>>> GetLeaderboard(
+        [FromQuery] Region? region,
+        [FromQuery] CharacterClass? characterClass
+    )
```
Comment (namidaka) [inline src/WebApi/Controllers/LeaderboardController.cs:24]: this solves the stylecop warning
```suggestion
    public Task<ActionResult<Result<IList<CharacterPublicViewModel>>>> GetLeaderboard(
        [FromQuery] Region? region,
        [FromQuery] CharacterClass? characterClass)
```

**Review by namidaka — CHANGES_REQUESTED** [review]

**Review by namidaka — COMMENTED** [review]
All good for me, what does @verdie-g  think?

**Comment by verdie-g** [comment]
I'll have a look tomorrow

File: src/Application/Characters/Queries/GetLeaderboardQuery.cs [hunk 3]
```diff
@@ -29,6 +32,7 @@ public async Task<Result<IList<CharacterPublicViewModel>>> Handle(GetLeaderboard
             var topRatedCharactersByRegion = await _db.Characters
                 .OrderByDescending(c => c.Rating.CompetitiveValue)
                 .Where(c => req.Region == null || req.Region == c.User!.Region)
+                .Where(c => req.CharacterClass == null || req.CharacterClass == c.Class)
```
Comment (verdie-g) [inline src/Application/Characters/Queries/GetLeaderboardQuery.cs:35]: You can merge that with the previous where clause

**Review by verdie-g — CHANGES_REQUESTED** [review]

**Review by verdie-g — APPROVED** [review]

**Pull request merged**